rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -46,7 +48,19 @@ Node::Node(const std::string & node_name, context::Context::SharedPtr context)
 : name_(node_name), context_(context),
   number_of_subscriptions_(0), number_of_timers_(0), number_of_services_(0)
 {
-  auto node = rmw_create_node(name_.c_str());
+  size_t domain_id = 0;
+  char * ros_domain_id = getenv("ROS_DOMAIN_ID");
+  if (ros_domain_id) {
+    unsigned long number = strtoul(ros_domain_id, NULL, 0);
+    if (number == (std::numeric_limits<unsigned long>::max)()) {
+      fprintf(stderr,
+        "Failed to interpret ROS_DOMAIN_ID '%s' as integral number\n", ros_domain_id);
```

Comment:
I thought about out and then ended up not putting a message in. It is perfectly fine for using it without - especially since that will be what every user does by default.